server/app/routes/locations/index.js

```diff
@@ -0,0 +1,54 @@
+/* jshint node:true*/
+'use strict';
+
+var db = require('../../../db');
+var Location = db.model('location');
+var router = require('express').Router();
+
+module.exports = router;
+
+router.param('id', function (req, res, next, theId) {
+    Location.findById(theId)
+        .then(function (foundLocation) {
+            if (!foundLocation) res.sendStatus(404);
+            else req.productInstance = foundLocation;
```

Comment:
Might be more clear if req.productInstance was req.locationInstance but everything works as it